src/components/IllustratedHeaderPageLayout.tsx

```diff
@@ -1,43 +1,34 @@
-import PropTypes from 'prop-types';
 import React from 'react';
 import useTheme from '@hooks/useTheme';
 import useThemeStyles from '@hooks/useThemeStyles';
 import HeaderPageLayout from './HeaderPageLayout';
-import headerWithBackButtonPropTypes from './HeaderWithBackButton/headerWithBackButtonPropTypes';
+import HeaderWithBackButtonProps from './HeaderWithBackButton/types';
 import Lottie from './Lottie';
+import DotLottieAnimation from './LottieAnimations/types';
 
-const propTypes = {
-    ...headerWithBackButtonPropTypes,
-
+type IllustratedHeaderPageLayoutProps = {
     /** Children to display in the lower half of the page (below the header section w/ an animation) */
-    children: PropTypes.node.isRequired,
+    children?: React.ReactNode;
 
-    /** The illustration to display in the header. Can be either an SVG component or a JSON object representing a Lottie animation. */
-    illustration: PropTypes.oneOfType([PropTypes.func, PropTypes.object]).isRequired,
+    /** The illustration to display in the header. Can be a JSON object representing a Lottie animation. */
+    illustration: DotLottieAnimation;
 
     /** The background color to apply in the upper half of the screen. */
-    backgroundColor: PropTypes.string,
-
-    /** A fixed footer to display at the bottom of the page. */
-    footer: PropTypes.node,
+    backgroundColor?: string;
 
     /** Overlay content to display on top of animation */
-    overlayContent: PropTypes.func,
-};
+    overlayContent?: () => React.ReactNode;
 
-const defaultProps = {
-    backgroundColor: undefined,
-    footer: null,
-    overlayContent: null,
+    /** Props to pass to HeaderWithBackButton */
+    propsToPassToHeader: HeaderWithBackButtonProps;
```

Comment:
Instead of defining `propsToPassToHeader` property, use & operator:
```ts
type IllustratedHeaderPageLayoutProps = HeaderWithBackButtonProps & {
...
}
```